### DISCUSSION TITLE
Gradient of the hydrostatic stress \sigma_H

### DISCUSSION THREAD

**Discussion by saturn00000** [idaholab/moose#22661] [Q&A Getting Started] [answered]
Dear MOOSE Team

can we define the **gradient** of the **hydrostatic stress**, \sigma_H, which is computed at the integration points from the nodal displacements in MOOSE?

The test input file looks like:

```
...

[AuxVariables]
  [hydrostatic]
    order = CONSTANT
    family = MONOMIAL
  []
  [grad_u_x]
    order = CONSTANT
    family = MONOMIAL
  []
[]

[AuxKernels]
  [hydrostatic]
    type = ADRankTwoScalarAux
    rank_two_tensor = stress
    variable = hydrostatic
    scalar_type = Hydrostatic
  []

  [grad_u_x_aux]
    type = VariableGradientComponent
    variable = grad_u_x
    component = x
    gradient_variable = hydrostatic
  []

[]
...
```
The results show that the value of the gradient of the hydrostatic stress in x direction (defined as grad_u_x) is 0...

**Comment by hugary1995** [comment, marked as answer]
You can do a local L2 projection which preserves dual numbers. See e.g. https://github.com/hugary1995/eel/blob/main/src/materials/chemistry/ChemicalPotential.C

**Reply by saturn00000** [reply]
can we just set ` order = second `

e.g.

```
  [hydrostatic]
    order = second 
    family = MONOMIAL
  []
```

**Reply by hugary1995** [reply]
Yes, actually you can get it to work using any monomial equal or above first order. Please be reminded that all derivatives in the ADReal will be lost by projecting it onto an AuxVariable. So if your model is somehow coupled to the gradient of the hydrostatic stress, the Jacobian will no longer be exact.
